sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -129,3 +135,11 @@ case class UserDefinedFunction protected[sql] (
     }
   }
 }
+
+object UserDefinedFunction {
```

Comment:
`private[sql]` since we don't explicitly mention `expressions` package is meant to be internal.

Reply:
I am okay if this is the only way.

Reply:
Here I'm adding back the public `unapply` method, so this must be public.